src/validator.js

```diff
@@ -75,6 +88,44 @@ export default class Validator {
     return output;
   }
 
+  scan(_Xpi=Xpi) {
+    return new Promise((resolve, reject) => {
+      this.checkFileExists(this.packagePath)
+        .then(() => {
+          this.xpi = new _Xpi(this.packagePath);
+          return this.xpi.getMetaData();
+        }).then(() => {
+          return this.xpi.getJSFiles();
+        })
+        .then((jsFiles) => {
+          jsFiles.forEach((filename) => {
+            this.xpi.getFileAsString(filename)
+              .then((code) => {
+                let jsScanner = new JavaScriptScanner(code, filename);
+                return jsScanner.scan();
+              })
+              .then((validatorMessages) => {
+                validatorMessages.forEach((message) => {
+                  this.addToCollector(message);
+                });
+
+                resolve(this.collector);
```

Comment:
I’ve been testing against things with one file, yeah. You’re right. This needs to be cleaned up, whoops.

- tofumatt

On 5 October 2015 at 16:34:30, Stuart Colville (notifications@github.com) wrote:

In src/validator.js:

> +        }).then(() => {
> +          return this.xpi.getJSFiles();
> +        })
> +        .then((jsFiles) => {
> +          jsFiles.forEach((filename) => {
> +            this.xpi.getFileAsString(filename)
> +              .then((code) => {
> +                let jsScanner = new JavaScriptScanner(code, filename);
> +                return jsScanner.scan();
> +              })
> +              .then((validatorMessages) => {
> +                validatorMessages.forEach((message) => {
> +                  this.addToCollector(message);
> +                });
> +
> +                resolve(this.collector);
Doesn't this mean you'll resolve on the first file - what about the following iterations.

—
Reply to this email directly or view it on GitHub.